src/main/java/io/temporal/samples/dsl/DynamicDslWorkflow.java

```diff
@@ -0,0 +1,165 @@
+/*
+ *  Copyright (c) 2020 Temporal Technologies, Inc. All Rights Reserved
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.samples.dsl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.serverlessworkflow.api.actions.Action;
+import io.serverlessworkflow.api.interfaces.State;
+import io.serverlessworkflow.api.states.EventState;
+import io.serverlessworkflow.api.states.OperationState;
+import io.serverlessworkflow.api.states.SwitchState;
+import io.serverlessworkflow.api.switchconditions.DataCondition;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.common.converter.EncodedValues;
+import io.temporal.workflow.ActivityStub;
+import io.temporal.workflow.DynamicSignalHandler;
+import io.temporal.workflow.DynamicWorkflow;
+import io.temporal.workflow.Workflow;
+import java.util.List;
+import org.slf4j.Logger;
+
+public class DynamicDslWorkflow implements DynamicWorkflow {
+
+  private io.serverlessworkflow.api.Workflow dslWorkflow;
+  private JsonNode workflowData;
+  private String workflowType;
+  private Logger logger = Workflow.getLogger(DynamicDslWorkflow.class);
+
+  @Override
+  public Object execute(EncodedValues args) {
+    // Get first input and convert to SW Workflow object
+    String dsl = args.get(0, String.class);
+    dslWorkflow = io.serverlessworkflow.api.Workflow.fromSource(dsl);
```

Comment:
I think we want to discourage passing the whole workflow definition for each workflow. I would pass only workflow DSL name which includes a version to support changes and then load it from an in-process cache. This would allow sharing a single `io.serverlessworkflow.api.Workflow` instance across all currently executing workflows. It is possible because this instance is immutable.

Reply:
fixed